File: Backend/Models/Prescription.cs
using System.ComponentModel.DataAnnotations;

namespace QuanLyBenhVien.API.Models;

public class Prescription
{
    [Key]
    public int PrescriptionID { get; set; }
    public int? RecordID { get; set; }
    public string? Diagnosis { get; set; }
    public DateTime? CreatedAt { get; set; }
    public string? Status { get; set; }

    // Navigation properties
    public virtual MedicalRecord? MedicalRecord { get; set; }
    public virtual ICollection<PrescriptionDetail> Details { get; set; } = new List<PrescriptionDetail>();
}

File: SpringBoot/src/main/java/com/hospital/qlbv/entity/Prescription.java
package com.hospital.qlbv.entity;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.*;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "prescriptions")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Prescription {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer prescriptionId;
    
    @Column(name = "record_id")
    private Integer recordId;
    
    private String diagnosis;
    
    @Column(name = "created_at")
    private String createdAt;
    
    private String status;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "record_id", insertable = false, updatable = false)
    @JsonIgnoreProperties({"prescriptions", "testRequests"})
    private MedicalRecord medicalRecord;
    
    @OneToMany(mappedBy = "prescription", cascade = CascadeType.ALL)
    @JsonIgnoreProperties({"prescription"})
    @Builder.Default
    private List<PrescriptionDetail> details = new ArrayList<>();
}
